pexv2/pd.go

```diff
@@ -22,6 +34,98 @@ type InputDescriptor struct {
 	Constraints Constraints `json:"constraints"`
 }
 
+type tokenizedField struct {
+	token string
+	path  string
+}
+
+// SelectCredentials selects vcJWTs based on the constraints defined in the input descriptor
+func (ind InputDescriptor) SelectCredentials(vcJWTs []string) ([]string, error) {
+	jsonSchema := JSONSchema{
+		Schema:     "http://json-schema.org/draft-07/schema#",
+		Type:       "object",
+		Properties: make(map[string]Filter, len(ind.Constraints.Fields)),
+		Required:   make([]string, 0, len(ind.Constraints.Fields)),
+	}
+
+	tokenizedFields := make([]tokenizedField, 0, len(ind.Constraints.Fields))
+	tokens := make([]string, len(ind.Constraints.Fields))
+
+	for i, field := range ind.Constraints.Fields {
+		token := strconv.FormatInt(time.Now().UnixNano(), 10)
+		tokens[i] = token
+
+		for _, path := range field.Path {
+			tf := tokenizedField{token: token, path: path}
```

Comment:
so i think this is a bit of a departure - my tokenField had token and []string path, but yours has 1:1 token to path

Reply:
Indeed it reduces a loop during actual eval